src/NeoSharp.Core/VM/StateMachine.cs

```diff
@@ -0,0 +1,354 @@
+            //    return false;
+            //if (stack.PeekByteArray().Length > 1024)
+            //    return false;
+            //string name = Encoding.UTF8.GetString(stack.PopByteArray());
+            //Fixed8 amount = new Fixed8((long)engine.CurrentContext.EvaluationStack.Pop().GetBigInteger());
+            //if (amount == Fixed8.Zero || amount < -Fixed8.Satoshi) return false;
+            //if (assetType == AssetType.Invoice && amount != -Fixed8.Satoshi)
+            //    return false;
+            //byte precision = (byte)engine.CurrentContext.EvaluationStack.Pop().GetBigInteger();
+            //if (precision > 8) return false;
+            //if (assetType == AssetType.Share && precision != 0) return false;
+            //if (amount != -Fixed8.Satoshi && amount.GetData() % (long)Math.Pow(10, 8 - precision) != 0)
+            //    return false;
+            //ECPoint owner = ECPoint.DecodePoint(stack.PopByteArray(), ECCurve.Secp256r1);
+            //if (owner.IsInfinity) return false;
+            //if (!CheckWitness(engine, owner))
+            //    return false;
+            //UInt160 admin = new UInt160(stack.PopByteArray());
+            //UInt160 issuer = new UInt160(stack.PopByteArray());
+            //AssetState asset = _assets.GetOrAdd(tx.Hash, () => new AssetState
+            //{
+            //    AssetId = tx.Hash,
+            //    AssetType = assetType,
+            //    Name = name,
+            //    Amount = amount,
+            //    Available = Fixed8.Zero,
+            //    Precision = precision,
+            //    Fee = Fixed8.Zero,
+            //    FeeAddress = new UInt160(),
+            //    Owner = owner,
+            //    Admin = admin,
+            //    Issuer = issuer,
+            //    Expiration = Blockchain.Default.Height + 1 + 2000000,
+            //    IsFrozen = false
+            //});
+            //engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(asset));
+            return true;
+        }
+
+        private bool Asset_Renew(IStackAccessor stack)
+        {
+            var asset = stack.Pop<Asset>();
+            if (asset == null) return false;
+
+            var years = (byte)stack.PopBigInteger();
+
+            asset = _assets.GetAndChange(asset.Id);
+
+            if (asset.Expiration < _blockchainContext.CurrentBlock.Index + 1)
+                asset.Expiration = _blockchainContext.CurrentBlock.Index + 1;
+
+            try
+            {
+                asset.Expiration = checked(asset.Expiration + years * 2000000u);
+            }
+            catch (OverflowException)
+            {
+                asset.Expiration = uint.MaxValue;
+            }
+
+            stack.Push(asset.Expiration);
+
+            return true;
+        }
+
+        private bool Contract_Create(IStackAccessor stack)
+        {
+            var script = stack.PopByteArray();
```

Comment:
This method call `Dispose` to the base StackItem?

Reply:
I'll probably switch to VM abstractions instead. But yes it's a good point.